Source/NSubstitute/Core/CallSpecificationFactory.cs

```diff
@@ -15,7 +15,7 @@ public ICallSpecification CreateFrom(ICall call, MatchArgs matchArgs)
         {
             var methodInfo = call.GetMethodInfo();
             var argumentSpecs = call.GetArgumentSpecifications();
-            var arguments = call.GetArguments();
+            var arguments = call.GetOriginalArguments();
```

Comment:
It changes behavior for all cases, including `Returns` and `When/Do` handlers as well - it doesn't matter much whether you modify args via `ICall` directly or `CallInfo`. 
I've created a [bug](https://github.com/AutoFixture/AutoFixture/issues/707) in AF in past which describes this case (currently AF works via `Returns` mechanism).